go/cmd/server/server_test.go

```diff
@@ -138,3 +170,149 @@ func TestGetOnlineFeaturesSqlite(t *testing.T) {
+
+		assert.Nil(t, err)
+		assert.Equal(t, len(values)-1 /*request id column not counted*/, len(expectedLogValues))
+		// Need to iterate through and compare because certain values in types.RepeatedValues aren't accurately being compared.
+		for name, val := range values {
+			if name == "RequestId" {
+				// Ensure there are request ids for each entity.
+				assert.Equal(t, len(val.Val), len(response.Results[0].Values))
+			} else {
+				assert.Equal(t, len(val.Val), len(expectedLogValues[name].Val))
+				for idx, featureVal := range val.Val {
+					assert.Equal(t, featureVal.Val, expectedLogValues[name].Val[idx].Val)
+				}
+			}
+
+		}
+	}
+	err = test.CleanUpLogs(logPath)
+	assert.Nil(t, err)
+}
+
+// Generate the expected log rows based on the resulting feature vector returned from GetOnlineFeatures.
+func GetExpectedLogRows(featureNames []string, results []*serving.GetOnlineFeaturesResponse_FeatureVector) (map[string]*types.RepeatedValue, [][]int32, [][]int64) {
+	numFeatures := len(featureNames)
+	numRows := len(results[0].Values)
+	featureValueLogRows := make(map[string]*types.RepeatedValue)
+	featureStatusLogRows := make([][]int32, numRows)
+	eventTimestampLogRows := make([][]int64, numRows)
+	for idx := 0; idx < len(results); idx++ {
+		valArray := make([]*types.Value, 0)
+		for row_idx := 0; row_idx < numRows; row_idx++ {
+			featureStatusLogRows[row_idx] = make([]int32, numFeatures)
+			eventTimestampLogRows[row_idx] = make([]int64, numFeatures)
+			valArray = append(valArray, results[idx].Values[row_idx])
+			featureStatusLogRows[row_idx][idx] = int32(serving.FieldStatus_PRESENT)
+			eventTimestampLogRows[row_idx][idx] = results[idx].EventTimestamps[row_idx].AsTime().UnixNano() / int64(time.Millisecond)
+
+		}
+		featureValueLogRows[featureNames[idx]] = &types.RepeatedValue{
+			Val: valArray,
+		}
+	}
+
+	return featureValueLogRows, featureStatusLogRows, eventTimestampLogRows
+}
+
+func CreateFeatureServiceInFeatureStore(fs *feast.FeatureStore) {
+	f1 := core.FeatureSpecV2{
+		Name:      "acc_rate",
+		ValueType: types.ValueType_FLOAT,
+	}
+	f2 := core.FeatureSpecV2{
+		Name:      "conv_rate",
+		ValueType: types.ValueType_FLOAT,
+	}
+	f3 := core.FeatureSpecV2{
+		Name:      "avg_daily_trips",
+		ValueType: types.ValueType_INT64,
+	}
+	projection1 := core.FeatureViewProjection{
+		FeatureViewName:      "driver_hourly_stats",
+		FeatureViewNameAlias: "",
+		FeatureColumns:       []*core.FeatureSpecV2{&f1, &f2, &f3},
+		JoinKeyMap:           map[string]string{},
+	}
+	featureServiceSpec := &core.FeatureServiceSpec{
+		Name:     "test_service",
+		Project:  "feature_repo",
+		Features: []*core.FeatureViewProjection{&projection1},
+	}
+	ts := timestamppb.New(time.Now())
+	featureServiceMeta := &core.FeatureServiceMeta{
+		CreatedTimestamp:     ts,
+		LastUpdatedTimestamp: ts,
+	}
+	featureService := &core.FeatureService{
+		Spec: featureServiceSpec,
+		Meta: featureServiceMeta,
+	}
+	registry := fs.Registry()
```

Comment:
so where this registry goes? is it used anywhere?

Reply:
It is used in feature Service right below, essentially, I cache the featureservice manually using the featureservice proto 
fs.Registry just returns a pointer and we need it to actually query the feature service from registry when the logging service queries for it.